components/scream/src/physics/rrtmgp/scream_rrtmgp_interface.hpp

```diff
@@ -2,25 +2,59 @@
 #define SCREAM_RRTMGP_INTERFACE_HPP
 
 #include "ekat/ekat_assert.hpp"
+#include "mo_gas_optics_rrtmgp.h"
+#include "mo_cloud_optics.h"
+#include "mo_fluxes.h"
 
 namespace scream {
     namespace rrtmgp {
         /* 
-         * These may or may not be needed to interface with the fortran code.
-         * If we jump directly to the C++ version of RRTMGP, or use the C++
-         * API for RRTMGP, we should be able to do away with these.
+         * Objects containing k-distribution information need to be initialized
+         * once and then persist throughout the life of the program, so we
+         * declare them here within the rrtmgp namespace.
          */
-        extern "C" {
-            void rrtmgp_init_f90();
-            void rrtmgp_main_f90();
-            void rrtmgp_finalize_f90();
-        }  // extern "C"
+        extern GasOpticsRRTMGP k_dist_sw;
+        extern GasOpticsRRTMGP k_dist_lw;
         /*
-         * Assuming we can jump directly to using a C++ API...
+         * Objects containing cloud optical property look-up table information.
+         * We want to initialize these once and use throughout the life of the
+         * program, so declare here and read data in during rrtmgp_initialize().
+         */
+        extern CloudOptics cloud_optics_sw;
+        extern CloudOptics cloud_optics_lw;
+        /*
+         * Initialize data for RRTMGP driver
+         */
+        extern void rrtmgp_initialize();
+        /*
+         * Main driver code to run RRTMGP
+         */
+        extern void rrtmgp_main(
+                real2d &p_lay, real2d &t_lay, real2d &p_lev, real2d &t_lev, 
+                GasConcs &gas_concs, real2d &col_dry,
+                real2d &sfc_alb_dir, real2d &sfc_alb_dif, real1d &mu0, 
+                real2d &lwp, real2d &iwp, real2d &real, real2d &rei,
+                real2d &sw_flux_up, real2d &sw_flux_dn, real2d &sw_flux_dn_dir,
+                real2d &lw_flux_up, real2d &lw_flux_dn);
+        /*
+         * Perform any clean-up tasks
          */
-        extern void rrtmgp_init();
-        extern void rrtmgp_main();
         extern void rrtmgp_finalize();
+        /*
+         * Things that probably should not be used outside of the RRTMGP interface...
```

Comment:
If you don't want to make these functions available to anyone outside of a certain context, you might consider placing them into a separate header only included by stuff within the interface.

Reply:
Yeah that's not a bad idea. I'm somewhat on the fence on this, but in that strategy, would you recommend putting these prototypes in a separate header and then the function bodies in a separate source file? I.e., something like
```
scream_rrtmgp_interface_private.hpp
scream_rrtmgp_interface_private.cpp
```

Reply:
That would certainly work. I don't think this is a big deal, but if you're worried about limiting visibility, it never hurts to move them out of sight.